### DISCUSSION TITLE
Which improvements do  you think are needed in OrientDB 4.0 ?

### DISCUSSION THREAD

**Discussion by andrii0lomakin** [orientechnologies/orientdb#10063] [Polls]
Hi, guys.

Please share which improvements the dev team should concentrate on in the 4.0 release.
Please feel free to add your features/improvements in the comments. Some will be added to the following pools.

We would greatly appreciate all of your responses and feedback.
Please don't hesitate to suggest any improvements or features for the new version.
 
This pool is the first in the series and provides information for the following more detailed pools.

This pool is focused solely on discussing the features and improvements of new OrientDB versions. All other types of comments will be removed.

**Comment by mdre** [comment]
It could be nice to have a way to draw a graph based on Class and edges so we can see the abstract graph like we do in a ERD.

This is a short example of relation in the database. There's 134 different class types.

![image](https://github.com/orientechnologies/orientdb/assets/5312780/14901a81-a6dd-4b2d-9393-978a27f8d60e)


but if I want a DER it should be something like this:

![image](https://github.com/orientechnologies/orientdb/assets/5312780/8d78a0c1-53df-4e9f-936c-9f3decbb84e2)


(I omit the name of relationships in the graph and should be added)
I think that if we can register some metadata in the edge, like (type) in (cardinality) - (cardinality) out (type) , the graph could be made.
and if a edge do not have that info it is ignored in the Graph-Relation-Explain graph
The relations should be visible even if no instance exist.

**Reply by andrii0lomakin** [reply]
That is a good idea, definitely worth considering. Thank you.

**Reply by andrii0lomakin** [reply]
Hi @mdre https://github.com/orientechnologies/orientdb/issues/10103 , the first step in this direction. Thank you for the idea.

**Reply by mdre** [reply]
That's great!

**Comment by andrii0lomakin** [comment]
Hi guys.

As I can see, improvements in storage durability are leading, so we have preliminarily planned several issues to improve durability in the 4.0 release. You can find them by looking for the `durability` label in the issue tracker. Nonetheless, that is only a preliminary decision. Please continue to vote so we can create a release that fits your interests as best as possible. All guys and girls who vote, please accept my sincere gratitude to all of you! 

The "lost" features will not be forgotten. They will be partially addressed in this release.

**Comment by adriancoca** [comment]
@laa  Hello and welcome back. Enterprise agent can be included in Orient Db by default ?https://repo1.maven.org/maven2/com/orientechnologies/agent/

**Reply by andrii0lomakin** [reply]
Yes, that is the plan to integrate it into an open-source version.

**Comment by andrii0lomakin** [comment]
We have added `write-performance` label for all related issues.

**Comment by andrii0lomakin** [comment]
Guys, again, thank you everyone for your feedback.
A separate post for a discussion of database durability was created: https://github.com/orientechnologies/orientdb/discussions/10113 . Everyone who voted for durability improvements please comment there.
